### PR TITLE
Ki/APPEALS-49727 - Task Actions: Other motion: Mark Task Complete

### PR DESCRIPTION
**Pull request by KiMauVA**
Resolves [APPEALS-49727 - Task Actions: Other motion: Mark Task Complete](https://jira.devops.va.gov/browse/APPEALS-49727)

# Description
Created CorrespondenceCompleteTaskModal that populates on Details page.  This pops up after "Mark task complete" has been selected on an Other Motion task, allowing the task to be Completed.

## Acceptance Criteria
- [X] Code compiles correctly

## Testing Plan
1. Navigate to a correspondence case review package page that does not have an intake already in progress(i.e. http://localhost:3000/queue/correspondence/053d5e15-716d-4bd7-8d7f-f6be737b52e3/review_package).
    1. NOTE: Save this URL(minus the review_package), you'll need it to navigate to the Correspondence Details after completing the intake.
2. Assign Correspondence Type.  Save changes.
3. Intake Correspondence with "Create Record" at bottom.
4. On Step 2, make sure you assign an "Other Motion" task to the correspondence.
5. Complete intake process.
6. Navigate to details page for that correspondence(i.e. http://localhost:3000/queue/correspondence/053d5e15-716d-4bd7-8d7f-f6be737b52e3/).
7. At the bottom of the first page, under the "Tasks not related to an appeal", find the Other Motion task you created.
8. Select "Mark task complete" under the dropdown action menu.
9. Click Cancel Link to test cancel.
10. Repeat dropdown.  Click X at top to test cancel.
11. Repeat dropdown.  Fill in the box on the Complete Modal and click "Mark Complete".
12. Task is removed from the "Tasks not related to an appeal" section, and a banner is displayed saying "Other motion task has been marked complete. If you have made a mistake, please email Litigation Support for any changes."

# Frontend
## User Facing Changes
 - [X] Screenshots of UI changes added to PR

### Front-End
Screenshot 2024-08-28 at 10 48 13 AM
Screenshot 2024-08-28 at 10 48 22 AM
Screenshot 2024-08-23 at 1 46 14 PM

### Dev/Supporting
Screenshot 2024-08-21 at 4 49 35 PM
Screenshot 2024-08-21 at 4 48 41 PM
Screenshot 2024-08-23 at 1 46 46 PM

### Test Coverage
Did you include any test coverage for your code? Check below:
- [X] RSpec
- [X] Jest (Updated Case Details jest issues)

### CODE REVIEW

File: app/models/tasks/mail_tasks_not_related_to_appeal/correspondence_mail_task.rb [hunk 1]
```diff
@@ -25,6 +25,7 @@ def self.create_child_task(parent, current_user, params)
     )
   end
 
+  # rubocop: disable Metrics/AbcSize
   def self.available_actions(user)
```
Comment (HunJerBAH) [inline app/models/tasks/mail_tasks_not_related_to_appeal/correspondence_mail_task.rb:29]: Outside of the scope of this ticket, but I think we have a bug @cacevesva where this needs to be **available_actions** instead of **self.available_actions**. I think we need a separate bug ticket to change this, but I think it's outside of the scope of this ticket.

File: app/models/tasks/mail_tasks_not_related_to_appeal/other_motion_correspondence_task.rb [hunk 2]
```diff
@@ -6,10 +6,14 @@ def label
...
-      Constants.TASK_ACTIONS.CANCEL_CORRESPONDENCE_TASK.to_h
+  def available_actions(user)
+    return [] unless user
+
+    options = [
+      Constants.TASK_ACTIONS.CANCEL_CORRESPONDENCE_TASK.to_h,
+      Constants.TASK_ACTIONS.COMPLETE_CORRESPONDENCE_TASK.to_h
     ]
+
+    options
```
Comment (HunJerBAH) [inline app/models/tasks/mail_tasks_not_related_to_appeal/other_motion_correspondence_task.rb:17]: Once the bug in correspondence_mail_task.rb is resolved, we can remove these available actions. Since this is in line with the AC of this ticket, I think we are good to keep them for now.

File: client/app/queue/components/CorrespondenceCompleteTaskModal.jsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+  // Additional properties - should be removed later once generic submit buttons are styled the same across all modals
+  const modalProps = {};
+
+  if ([
+    'AssessDocumentationTask',
+    'EducationAssessDocumentationTask',
+    'HearingPostponementRequestMailTask'
+  ].includes(task?.type) || task?.appeal.hasCompletedSctAssignTask) {
+    modalProps.submitButtonClassNames = ['usa-button'];
+  }
```
Comment (HunJerBAH) [inline client/app/queue/components/CorrespondenceCompleteTaskModal.jsx:72]: Is this **if** block needed or is this related to the appeals modal work?

**Review by HunJerBAH — DISMISSED** [review]

File: client/app/queue/components/CorrespondenceCompleteTaskModal.jsx (hunk 3 above)
Comment (KiMauVA) [reply client/app/queue/components/CorrespondenceCompleteTaskModal.jsx]: Unneeded.  Removed.

**Review by HunJerBAH — APPROVED** [review]

**Review by cacevesva — APPROVED** [review]
